events-collector/events-collector-server/src/db/models.rs

```diff
@@ -0,0 +1,173 @@
+//! Events collector database models
+
+use std::time::SystemTime;
+
+use bigdecimal::BigDecimal;
+use diesel::prelude::*;
+use uuid::Uuid;
+
+/// A wallet creation record
+#[derive(Insertable)]
+#[diesel(table_name = crate::db::schema::wallet_creations)]
+#[diesel(check_for_backend(diesel::pg::Pg))]
+pub struct WalletCreation {
+    /// The event ID
+    pub event_id: Uuid,
+    /// The event timestamp
+    pub event_timestamp: SystemTime,
+
+    /// The ID of the wallet that was created
+    pub wallet_id: Uuid,
+}
+
+/// A deposit record
+#[derive(Insertable)]
+#[diesel(table_name = crate::db::schema::deposits)]
+#[diesel(check_for_backend(diesel::pg::Pg))]
+pub struct Deposit {
+    /// The event ID
+    pub event_id: Uuid,
+    /// The event timestamp
+    pub event_timestamp: SystemTime,
+
+    /// The ID of the wallet that made the deposit
+    pub wallet_id: Uuid,
+    /// The mint of the asset that was deposited
+    pub mint: String,
+    /// The amount of the deposit
+    pub amount: BigDecimal,
```

Comment:
can we call out here whether this is decimal adjusted or not? and same for other fields that specify an erc20 amount